src/formatDate.js

```diff
@@ -8,7 +8,35 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const delimiters = ['-', '/', '.'];
+  const inputDelimiter = delimiters.find((d) => date.includes(d));
+  const dateParts = date.split(inputDelimiter);
+
+  const dateObj = {};
+
+  fromFormat.forEach((part, index) => {
+    dateObj[part] = dateParts[index];
+  });
+
+  if (dateObj['YYYY'] && toFormat.includes('YY')) {
+    dateObj['YY'] = dateObj['YYYY'].slice(2);
+  } else if (dateObj['YY'] && toFormat.includes('YYYY')) {
+    const year = parseInt(dateObj['YY'], 10);
+
+    dateObj['YYYY'] = year < 30 ? '20' + dateObj['YY'] : '19' + dateObj['YY'];
```

Comment:
The conversion from 'YY' to 'YYYY' does not account for the full range of years that could be represented by two digits. Consider adding a more robust handling for years outside the 20th and 21st centuries.